features/collections.feature

```diff
@@ -19,3 +19,15 @@ Feature: Collections
     When I choose radio button "Teaching and leadership" and click on "Continue"
     And I click on the button "Continue"
     Then I should see "How often do you want to get emails?"
+
+  Scenario: Check the feedback component loads
+    When I visit "/government/organisations"
+    And I confirm it is rendered by "collections"
+    And I click to report a problem with the page
+    Then I see the report a problem form
+    When I close the open feedback form
+    And I click to say the page is not useful
+    Then I see the email survey signup form
+    When I close the open feedback form
+    And I click to say the page is useful
+    Then I see the feedback confirmation message
```

Comment:
This repetition feels a bit yuck, but I'm not sure what (if anything?) can be done about it. You've followed your own rules of splitting tests by application.

It just makes me wonder if there's a less repetitive way for us to test features that span multiple applications because they exist in the global GOV.UK page layout. I suppose this could be considered to be similar to `Scenario: Check the crown logo links to GOV.UK homepage` in that regard.

Reply:
I agree it's yuck - more visibly yuck, at least.

First I need to make a correction: https://github.com/alphagov/smokey/pull/973/commits/0091108011cb5b95440798f975af57e04194fc3b was a poor change because this is not secondary, nor is it about site-wide config; it really is targeting communication with Static and I should revert that change.

Based on [the original commit](https://github.com/alphagov/smokey/commit/ca008962fe89e1b62a5df97a47b0db020670a542), the need for this test is likely to go away with Static: if we still really want to test this, it will be possible to do via the component and / or individual apps.

I'd like to explore a partial deletion option: is the data transfer with Static for all of these apps actually a critical part of GOV.UK as a whole? I don't think this is true for something like info-frontend, for example.

I'm wondering if we could cut this down to just:

- government-frontend
- frontend
- finder-frontend
- collections
- smartanswers

That would cut out half the repetition. Does that change anything for you? This is one of those classic testing problems where we've DRYed-up the code but still need tests for each place it's used.

Reply:
Sorry for being the one who introduced this pain point - I recall adding to Smokey being pretty much a last resort effort to add testing for something that wasn't testable in app. I'd agree it's not ideal to be a Smokey thing and would be much better served in the app, with Static being the pain point. I'm not quite sure why it can't be pretty much all handled in testing of the component gem but it's clearly way more complex than that now.

I guess the first thing I'd suggest for this is shall we explore making a step that can encapsulate the duplication. e.g:

```
  Scenario: Check the feedback component loads
    When I visit "/government/organisations"
    And I confirm it is rendered by "collections"
    Then I can operate with the feedback component
```

I think that'd make it much less painful to test.

> I'd like to explore a partial deletion option: is the data transfer with Static for all of these apps actually a critical part of GOV.UK as a whole? I don't think this is true for something like info-frontend, for example.

You're right. info-frontend get's so little love that it wouldn't be a biggie. Though it feels like for almost everything else we probably do care somewhat. E.g. email-alert-frontend, you can't sign in we kinda want people to be able to feedback, service-manual-frontend, 🤷. I worry though that partial deletion leaves it as rather ambiguous.